src/main/java/software/amazon/smithy/lsp/document/DocumentParser.java

```diff
@@ -704,4 +704,57 @@ private void nextNonWsNonComment() {
     private void reset() {
         rewind(0, 1, 1);
     }
+
+    /**
+     * Finds a contiguous range of non-whitespace characters starting from the given SourceLocation.
+     * If the sourceLocation happens to be a whitespace character, it returns a Range representing that column.
+     *
+     * Here is how it works:
+     * 1. We first jump to sourceLocation. If we can't, we return null.
+     * 2. We then check if the sourceLocation is a whitespace character. If it is, we return that column.
+     * 3. We then find the start of the contiguous range by traversing backwards until a whitespace character is found.
+     * 4. We then find the end of the contiguous range by traversing forwards until a whitespace character is found.
+     *
+     * @param sourceLocation The starting location to search from.
+     * @return A Range object representing the contiguous non-whitespace characters,
+     *         or null if not found.
+     */
+    public Range findContiguousRange(SourceLocation sourceLocation) {
+        if (!jumpToSource(sourceLocation)) {
+            return null;
+        }
+
+        int startLine = sourceLocation.getLine() - 1;
+        int startColumn = sourceLocation.getColumn() - 1;
```

Comment:
```suggestion
        Position startPosition = LspAdapter.toPosition(sourceLocation);
        int startLine = startPosition.getLine();
        int startColumn = startPosition.getCharacter();
```
`toPosition` takes care of the conversion to 0-based indexing. I had forgotten about it

Reply:
Thats true. I wish we had value types. Main concern from me is I don't want to accidentally forget about the `-1` if we refactor this later. What you could do, since you're creating a position anyways in either return path, is just re-use whatever `toPosition` gave you and call `setCharacter` if necessary.

Reply:
But I guess we will have the tests in DocumentParser for SourceLocation of (1,1), so up to you.

Reply:
yes I agree with the consistency logic. Just wanted to make sure that there is no other reason that I might be failing to understand. Will update.